packages/host/app/components/operator-mode/code-submode/playground/instance-chooser-dropdown.gts

```diff
@@ -151,140 +159,154 @@ interface Signature {
+        </CardContainer>
+      </BoxelSelect>
+    {{/if}}
 
-  <style scoped>
-    .loading-icon {
-      height: var(--boxel-form-control-height);
-    }
-    .instance-chooser {
-      height: 26px;
-      border: 1px solid var(--boxel-dark);
-      outline: none;
-    }
-    .instance-chooser :deep(.boxel-trigger-content) {
-      overflow: hidden;
-    }
-    :deep(
-      .boxel-select__dropdown .ember-power-select-option[aria-current='true']
-    ),
-    :deep(.instances-dropdown-content .ember-power-select-option) {
-      background-color: var(--boxel-light);
-      flex-wrap: nowrap;
-    }
-    :deep(.ember-power-select-option:hover .card) {
-      background-color: var(--boxel-100);
-    }
-    :deep(.boxel-trigger-content) {
-      font: var(--boxel-font-xs);
-    }
-    .instance-chooser :deep(.boxel-trigger-content) {
-      overflow: hidden;
-    }
-    .card,
-    .field {
-      height: 40px;
-      width: 375px;
-      max-width: 100%;
-      container-name: fitted-card;
-      container-type: size;
-      background-color: var(--boxel-light);
+    <style scoped>
+      .loading-icon {
+        height: var(--boxel-form-control-height);
+      }
+      .instance-chooser {
+        height: 26px;
+        border: 1px solid var(--boxel-dark);
+        outline: none;
+      }
+      .instance-chooser :deep(.boxel-trigger-content) {
+        font: var(--boxel-font-xs);
+        overflow: hidden;
+      }
+      :deep(
+        .boxel-select__dropdown .ember-power-select-option[aria-current='true']
+      ),
+      :deep(.instances-dropdown-content .ember-power-select-option) {
+        background-color: var(--boxel-light);
+        flex-wrap: nowrap;
+      }
+      :deep(.ember-power-select-option:hover .card) {
+        background-color: var(--boxel-100);
+      }
+      .card,
+      .field {
+        height: 40px;
+        width: 375px;
+        max-width: 100%;
+        container-name: fitted-card;
+        container-type: size;
+        background-color: var(--boxel-light);
+      }
+      .field {
+        padding: var(--boxel-sp-xs);
+        text-overflow: ellipsis;
+        white-space: nowrap;
+      }
+    </style>
+  </template>
+
+  @service private declare playgroundPanelService: PlaygroundPanelService;
+
+  private findSelectedCard = (prerenderedCards?: PrerenderedCard[]) => {
+    if (!prerenderedCards?.length) {
+      return;
     }
-    .field {
-      padding: var(--boxel-sp-xs);
-      text-overflow: ellipsis;
-      white-space: nowrap;
+
+    if (!this.args.selection?.card) {
+      let s = this.playgroundPanelService.peekSelection(this.args.moduleId);
+      if (!s) {
+        // if there's no selected card, choose the most recent card as selected
+        this.playgroundPanelService.persistSelections(
+          this.args.moduleId,
+          trimJsonExtension(prerenderedCards[0].url),
+          'isolated',
+          undefined,
+        );
+      }
+      return prerenderedCards[0];
```

Comment:
This is where we're persisting and returning the first available card if there was no selected card. Attempting to call `this.args.onSelect(prerenderedCards[0])` causes revalidation error. Using `this.playgroundPanelService.getSelection` causes revalidation error. That is why I added the `peekSelection` to the panel service.